### ISSUE TITLE
BatchNormalization layer has different output when loading a checkpoint saved from TF 2.0b1

### ISSUE THREAD

**Issue by edoardogiacomello** [tensorflow/tensorflow#38878]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): Yes
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04): Official Tensorflow Docker Container (2.0.0, 2.0.0-beta1, latest) with py3 and gpu support
- TensorFlow installed from (source or binary): Docker
- TensorFlow version (use command below): 2.0
- Python version: 3
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version: 10.0 / 10.2
- GPU model and memory: Tested on RTX 2080Ti, Titan V, GTX 1080

**Describe the current behavior**
In my project I trained several (adversarial) segmentation models using both Tensorflow 2.0.0-a0 and 2.0.0-b1. 
Upon upgrading Tensorflow to version 2.1 I noticed a decrease in the accuracy of my models between 5~15% (Dice Score) with respect to the version 2.0.0-b1. 
I ran several tests loading the same checkpoints in multiple tensorflow versions by comparing the layer outputs and I noticed that starting from the first BatchNormalization layer the outputs of networks loaded with TF 2.0.0 (stable) are different from those loaded using TF <= 2.0.0-b1(even though the loaded weights are the same). BatchNormalization behavior is consistent starting from 2.0 stable onward. 

**Describe the expected behavior**
The output of BatchNormalization layer (and thus of the network) should be the same when using TF 2.0.0b1 and 2.0.0, when the checkpoint has been trained and saved using beta1.

**Standalone code to reproduce the issue**
I tried reproducing the issue using the examples or pretrained networks, but the differences are not noticeable as they are in my models. I wrote a notebook that can be run on both 2.0.0b1 and 2.1 to highlight the differences. 
https://drive.google.com/open?id=1vw3vaPU32lqI2jAOvp0toXIgD84R_Tx4
Besides the notebook there are also the input data I used for testing and the output of the last layer for both TF2.0-b1 and TF2.1 (other versions are not inclued as they seem to match with either b1 or 2.1). I'm not including individual layers outputs due to size constraints.
In the pics fodler can find also some visual examples of the network outputs.

**Other info / logs** 
I'm attaching just an example of just one mismatching output for the same input. In the drive folder you can find the test for an entire batch.
![hist-61-2 0 0-beta1](https://user-images.githubusercontent.com/9888197/80245680-69085d80-866b-11ea-8e26-ee7fd8797f0f.jpg)
![hist-61-2 1 0](https://user-images.githubusercontent.com/9888197/80245684-69a0f400-866b-11ea-982e-1722c6733861.jpg)

I couldn't understand what this could be related to, I'd like to have some insights of what could be the cause.
Thank you!

**Comment by amahendrakar** [comment]
@edoardogiacomello,
I tried to reproduce the issue but did not observe much difference in the output with [TF v2.0](https://colab.research.google.com/gist/amahendrakar/65b2c0ba8212dc311fba6dfe8c4c0619/38878-2-0.ipynb), [TF v2.1](https://colab.research.google.com/gist/amahendrakar/160fce6364808f672b111819b2ad8271/38878-2-1.ipynb) and [TF v2.2.0-rc3](https://colab.research.google.com/gist/amahendrakar/bdee9985cb66c522c2ad0e30e76f7bac/38878-2-2.ipynb). Please find the attached gist. Thanks!

**Comment by edoardogiacomello** [comment]
> @edoardogiacomello,
> I tried to reproduce the issue but did not observe much difference in the output with [TF v2.0](https://colab.research.google.com/gist/amahendrakar/65b2c0ba8212dc311fba6dfe8c4c0619/38878-2-0.ipynb), [TF v2.1](https://colab.research.google.com/gist/amahendrakar/160fce6364808f672b111819b2ad8271/38878-2-1.ipynb) and [TF v2.2.0-rc3](https://colab.research.google.com/gist/amahendrakar/bdee9985cb66c522c2ad0e30e76f7bac/38878-2-2.ipynb). Please find the attached gist. Thanks!

I'm sorry, I may have made a mistake while managing the different containers. I checked again and I've been able to reproduce the error in version **2.0.0-beta1** (both on Docker and Colab), so the change may have happened between **2.0.0b1** and **2.0**. Starting from 2.0 (stable) the results are consistent with future versions, as you showed.

I'd like to find out what's causing this difference in BatchNormalization outputs and if there's a way to adapt my checkpoints to the stable version without a complete re-training.

**Comment by jvishnuvardhan** [comment]
@edoardogiacomello Please feel free to close the issue as this was resolved already. You could use stable version of `TF2.1` or `TF2.0`. Thanks!

**Comment by edoardogiacomello** [comment]
@jvishnuvardhan BatchNormalization layer may have been fixed (it changed for sure), but this issue is referred to loading a model that is trained with a previous version and it doesn't behave the way it should. I don't see how using a stable version could fix the issue, which is indeed related to loading a beta checkpoint in a stable version. 

Could you give a reference on what the issue was or how it has been resolved? I'd like to avoid re-launching about 8 months of training (or getting stuck into beta for the whole project) if there could be a workaround. Thank you :)

**Comment by karmel** [comment]
There were a number of changes between the beta and 2.1, and we would recommend retraining or adjusting weights as necessary. For long-lived models, it is best to use stable versions of TF, as nightlies and RCs sometimes contain issues that require non-backwards compatible changes versus the subsequent release.
